### PR TITLE
Provides a skeleton express server connected to a hosted mongodb

### PR DESCRIPTION
**Pull request by haletothewood**
This PR resolves issue #1 and introduces a Backend directory containing a node.js express server that is connected to an mLab hosted mongodb.

It also introduces a secrets file that is necessary to source control private config such as passwords and usernames.

There is no functionality at this stage just the scaffolding to build routes.

For more information on node.js express see the [documentation](https://expressjs.com/en/api.html#app). 

express-hello-world

### CODE REVIEW

File: Backend/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+Download [keepassX](https://www.keepassx.org/downloads) (You may have to allow for third-party applications in your settings)
+
+Open `secrets.kdbx` and follow the instructions in the file `mlab db ido-development` under the development group.
+
+
+### Run
+
+```bash
+$ npm install
```
Comment (ainsleybc) [inline Backend/README.md]: npm ci && npm start (keep it short!) 🚀

File: Backend/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+$ npm start
+
+> ido.backend@0.0.1 start <PATH>
```
Comment (ainsleybc) [inline Backend/README.md]: I wouldn't stick lines 22-25 in the readme - it's not very useful info.
Instead, add a line to explain that the server will be running on port x

Comment (haletothewood) [reply Backend/README.md]: Yeah, fair point, I didn't have it originally but decided to put it in last minute. Happy to remove.

File: Backend/README.md (hunk 1 above)
Comment (haletothewood) [reply Backend/README.md]: I was going to recommend introducing `npm ci` when we integrate CI/CD but there's no harm to add it now.

Comment (haletothewood) [reply Backend/README.md]: For reference: https://blog.npmjs.org/post/171556855892/introducing-npm-ci-for-faster-more-reliable

Comment (ainsleybc) [reply Backend/README.md]: it's better for dependency management locally as well (works more like yarn and only installs from the lock file). npm install is flakey

Comment (haletothewood) [reply Backend/README.md]: No more `rm -rf node_modules` - works for me.

**Comment by haletothewood** [comment]
@ainsleybc Changes made, ready for review.

File: Backend/package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  "devDependencies": {
+    "eslint": "^5.7.0",
+    "eslint-config-airbnb": "^17.1.0",
+    "eslint-config-standard": "^12.0.0",
+    "eslint-plugin-import": "^2.14.0",
+    "eslint-plugin-jsx-a11y": "^6.1.2",
+    "eslint-plugin-node": "^7.0.1",
+    "eslint-plugin-promise": "^4.0.1",
+    "eslint-plugin-react": "^7.11.1",
```
Comment (ainsleybc) [inline Backend/package.json]: are all these plugins necessary??

File: Backend/package.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  "name": "ido.backend",
+  "version": "0.0.1",
+  "description": "An express API server for iDo Frontend",
+  "main": "server.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/haletothewood/iDO.git"
```
Comment (ainsleybc) [inline Backend/package.json]: code-collective... 💩

File: Backend/package.json (hunk 3 above)
Comment (haletothewood) [reply Backend/package.json]: It came with the 'Standard' template? We could simplify and just create our own but this was the easiest option.

File: Backend/package.json (hunk 4 above)
Comment (haletothewood) [reply Backend/package.json]: Yep, forgot to go back and update this. I may actually just rebase the whole branch.

File: Backend/server.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import express from 'express'
+import { MongoClient } from 'mongodb'
+import { url } from './database/db.js'
+
+require('dotenv').config({ path: '../.env' })
```
Comment (jonsanders101) [inline Backend/server.js]: Does using dotenv effectively replace passing in environment variables via the command line when starting up the app?

Comment (haletothewood) [reply Backend/server.js]: [Yes](https://www.npmjs.com/package/dotenv)

Comment (haletothewood) [reply Backend/server.js]: I'm open to other implementations. This is all about us figuring out how we want to develop this.

Comment (jonsanders101) [reply Backend/server.js]: No, this looks good. I had read through the npm package but I wasn't sure from it whether that meant you can't pass in environment variables. Having a single file for them is a gooooood solution.

Comment (haletothewood) [reply Backend/server.js]: 👍 Apologies if the 'Yes' came across passive aggressive! Not intended.

**Comment by haletothewood** [comment]
@Ainsley, changed made. What are your thoughts on the plug-ins...keep using the 'Standard' config or strip them away and use our own config.

**Pull request merged**